extensions/ipynb/src/notebookSerializer.ts

```diff
@@ -70,11 +71,46 @@ export class NotebookSerializer implements vscode.NotebookSerializer {
 		return data;
 	}
 
-	public serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Uint8Array {
-		return new TextEncoder().encode(this.serializeNotebookToString(data));
+	public async serializeNotebook(data: vscode.NotebookData, _token: vscode.CancellationToken): Promise<Uint8Array> {
+		return new TextEncoder().encode(await this.serializeNotebookToString(data));
 	}
 
-	public serializeNotebookToString(data: vscode.NotebookData): string {
+	private async serializeViaWorker(workerData: notebookSerializationWorkerData): Promise<string> {
+		const workerThreads = await import('node:worker_threads');
+		const path = await import('node:path');
+		const { Worker } = workerThreads;
+
+		return await new Promise((resolve, reject) => {
+			const workerFile = path.join(__dirname, 'notebookSerializerWorker.js');
+			const worker = new Worker(workerFile, { workerData });
```

Comment:
@rebornix I think we might want to send the bytes using a similar technique like in `stringifyJsonWithBufferRefs`
Also workers have better perf when sending bytes across as opposed to strings (e.g. images and the like).
I think the larger the outputs get, and more complex (different tyeps of mime types), then the benefits would out weight not using it (after all we're using it in rpc layer).

Once we do that we can send the `vscode.NotebookData` object across directly and just import the types in worker and move 90% of the code from this file into a common place so it can be used in both workers as well as extension host context.